portal-kernel/src/com/liferay/portal/kernel/security/permission/propagator/PermissionPropagator.java

```diff
@@ -28,4 +28,9 @@ public void propagateRolePermissions(
 			long[] roleIds)
 		throws PortalException;
 
+	public void propagateRolePermissions(
+			long groupId, long companyId, String className, String primKey,
+			String childClassName, String childPrimKey, long[] roleIds)
```

Comment:
Should be `String parentClassName, long parentPrimKey, String childClassName, long childPrimKey`?